core/src/main/java/org/elasticsearch/plugins/PluginManagerCliParser.java

```diff
@@ -166,19 +168,28 @@ public ExitStatus execute(Settings settings, Environment env) throws Exception {
         private static final String NAME = "install";
 
         private static final CliToolConfig.Cmd CMD = cmd(NAME, Install.class)
-                .options(option("u", "url").required(false).hasArg(true))
                 .options(option("t", "timeout").required(false).hasArg(false))
                 .build();
 
         static Command parse(Terminal terminal, CommandLine cli) {
             String[] args = cli.getArgs();
+
+            // install [plugin-name/url]
             if ((args == null) || (args.length == 0)) {
-                return exitCmd(ExitStatus.USAGE, terminal, "plugin name is missing (type -h for help)");
+                return exitCmd(ExitStatus.USAGE, terminal, "plugin name or url is missing (type -h for help)");
             }
-
             String name = args[0];
+
+            URL pluginUrl = null;
+            // try provided name as URL,
+            try {
+                pluginUrl = new URL(name);
+                name = null;
+            } catch (MalformedURLException e) {
+                // ignore
```

Comment:
If we do `plugin install [nameOrUrl] and the argument doesn't parse to a valid url, we go on treating it as on of the other three options of a plugin name (e.g. analysis-icu, user/repo etc...). I can update the comment to make this clearer.